### PR TITLE
#MAN-55 Refactor Hours implementation

### PR DESCRIPTION
**Pull request by cdoyle-temple**
#MAN-55 Refactor Hours implementation

- move the logic into a service Object
- Call the service object from Rake
- Allow the service object to be called from a Admin Dashboard.
- Add tests for the service object

********************************************

Update hours controller and views
Move google logic to service
Add sync button to new hours dashboard
Correct singularization of LibraryHours to Library Hour (helpers and tests)

Everything except the test on the service

### CODE REVIEW

File: app/controllers/admin/library_hours_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module Admin
+  class LibraryHoursController < Admin::ApplicationController
+    def sync
+      SyncService::LibraryHours.call
+      flash[:notice] = "Hours synced"
```
Comment (bibliotechy) [inline app/controllers/admin/library_hours_controller.rb]: You need to handle the error case where sync service didn't work.

File: app/dashboards/library_hour_dashboard.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
```
Comment (bibliotechy) [inline app/dashboards/library_hour_dashboard.rb]: I'm not sure we want this dashboard at all. No one should be editing hours here. I know it exists just to have a sync button. Maybe what we need is a completely overridden Dashboard page that just has a big sync button and lists the last time a sync was run (based on the date with the most recent update_date time?

Comment (bibliotechy) [reply app/dashboards/library_hour_dashboard.rb]: Not going to block this PR for that, but going to create a new issue based off this idea.

File: spec/models/library_hour_spec.rb [hunk 3]
```diff
@@ -32,7 +32,7 @@
         library_hour = FactoryBot.create(:library_hour, k => v.first)
         library_hour.update(k => v.last)
         library_hour.save!
-        expect(library_hour.versions.last.changeset[k]).to match_array(v)
+        # expect(library_hour.versions.last.changeset[k]).to match_array(v)
```
[the remark this answers is not in the document]
Comment (cdoyle-temple) [reply spec/models/library_hour_spec.rb]: We're not versioning hours. It's not needed and should be removed.

**Pull request merged**